### ISSUE TITLE
Make gzipping of user data optional

### ISSUE THREAD

**Issue by jbg** [exoscale/terraform-provider-exoscale#215]
The provider currently gzips `user_data` passed to the `exoscale_compute_instance` resource before passing it (base64-encoded) to the API.

The Exoscale [API documentation] doesn't require that the user data is gzipped, it only requires it to be base64-encoded. So, at the moment, it's not possible to get plain-text user data via the Terraform provider, but it is possible via the API.

It would be useful if the compression of the user data could be disabled (e.g. `gzip_user_data = false`).

**Comment by alexandrevoilab** [comment]
This is a regression introduced by commit 5b5fc3363843e16b3c9510ad443bd2773acb587f .

Previously the provider configuration [gzip_user_data](https://registry.terraform.io/providers/exoscale/exoscale/latest/docs#gzip_user_data)  would allow one to disable gzip compression of `user_data`.
That option is still present, but no longer implemented in the provider's code.

This makes CoreOS / Flatcar pretty much unusable since [Ignition doesn't support gziped user_data](https://github.com/coreos/ignition/pull/1100#issuecomment-702364210).

I would gladly work on it and open a pull request, but I feel that there is some discussion to have on where that option really need to reside.
Maybe it should be an `exoscale_compute_instance` option? Since not all our instances are gzip intolerant.

I'll add that since the regression, a refactoring moved the impacted code to a new function, `encodeUserData`, inside `util.go`, from which I have no idea how to access the provider's `BaseConfig`.

**Comment by alexandrevoilab** [comment]
As a workaround, we can pass an already base64 encoded value to `user_data`.

Which will prevent it from being gziped.
https://github.com/exoscale/terraform-provider-exoscale/blob/8428c58c814b2bb30d0013cbaaf01744ffeeb543/exoscale/util.go#L99-L103

Exemple:
```
user_data = base64encode(data.ignition_config.mysql-ign-config.rendered)
```

**Comment by jessicatoscani** [comment]
It was fixed in #278, release in [0.51](https://github.com/exoscale/terraform-provider-exoscale/blob/v0.51.0/CHANGELOG.md#0510-august-9-2023)
